src/libs/actions/IOU.js

```diff
@@ -2936,13 +2967,24 @@ function approveMoneyRequest(expenseReport) {
         },
     ];
 
+    if (!_.isNull(currentNextStep)) {
+        optimisticData.push(optimisticNextStepsData);
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.NEXT_STEP}${expenseReport.reportID}`,
+            value: currentNextStep,
+        });
+    }
```

Comment:
I think it could be cleaner to define the optimistic object here too but not a blocker

Reply:
Done - 5b6700c.